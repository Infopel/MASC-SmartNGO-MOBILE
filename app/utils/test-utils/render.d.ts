export type RenderOptions = {
  wrapper?: React.ComponentType<any>
  createNodeMock?: (element: React.ReactElement) => any
}
export type AllTheProvidersProps = { children: React.ReactNode }
