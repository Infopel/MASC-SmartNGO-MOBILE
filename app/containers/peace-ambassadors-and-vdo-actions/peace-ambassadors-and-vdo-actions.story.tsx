import * as React from "react"
import { storiesOf } from "@storybook/react-native"
import { StoryScreen, Story, UseCase } from "../../../storybook/views"
import { color } from "../../theme"
import { PeaceAmbassadorsAndVdoActions } from "./peace-ambassadors-and-vdo-actions.component"

storiesOf("PeaceAmbassadorsAndVdoActions", module)
  .addDecorator((fn) => <StoryScreen>{fn()}</StoryScreen>)
  .add("Style Presets", () => (
    <Story>
      <UseCase text="Primary" usage="The primary.">
        <PeaceAmbassadorsAndVdoActions style={{ backgroundColor: color.error }} />
      </UseCase>
    </Story>
  ))
