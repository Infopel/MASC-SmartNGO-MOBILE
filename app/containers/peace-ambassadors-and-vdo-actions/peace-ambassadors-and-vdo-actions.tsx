import { withDatabase } from "@nozbe/watermelondb/DatabaseProvider"
import withObservables from "@nozbe/with-observables"
import { PeaceAmbassadorsAndVdoActionsListItem } from "components/peace-ambassadors-and-vdo-actions-list-item/peace-ambassadors-and-vdo-actions-list-item"
import { translate } from "i18n/translate"
import { observer } from "mobx-react-lite"
import { Divider, Fab, FlatList, VStack } from "native-base"
import { navigate } from "navigators/navigation-utilities"
import * as React from "react"
import { StyleProp, ViewStyle } from "react-native"
import { findAllEventsFromIncubator, IPeaceEmbassadorVDOAction } from "storage/queries/initiative"
import { PeaceAmbassadorsAndVdoActionsStyles as styles } from "./peace-ambassadors-and-vdo-actions.styles"

export interface PeaceAmbassadorsAndVdoActionsProps {
  /**
   * An optional style override useful for padding & margin.
   */
  style?: StyleProp<ViewStyle>
  items: IPeaceEmbassadorVDOAction[]
  initiativeId: string
}

/**
 * Describe your component here
 */
const Container = observer(function PeaceAmbassadorsAndVdoActions(props: PeaceAmbassadorsAndVdoActionsProps) {
  const { style, items, initiativeId } = props
  const containerStyle = {
    ...styles.container,
    ...(style && typeof style === "object" ? style : {}),
  }

  return (
    <VStack flex='1'>
      <FlatList data={items} ItemSeparatorComponent={() => <Divider mb='2' />} renderItem={({ item }) => <PeaceAmbassadorsAndVdoActionsListItem item={item} />} />
      <Fab renderInPortal={false} label={(translate('peace-ambassadors-and-vdo-details.add-action'))} onPress={() => navigate('form', { formId: 'pa-or-vdo-action', contextId: initiativeId })} />
    </VStack>
  )
})

export const PeaceAmbassadorsAndVdoActionsContainer = withDatabase(withObservables(['database', "initiativeId"], ({ database, initiativeId }) => ({ items: findAllEventsFromIncubator(database, initiativeId) }))(Container))
