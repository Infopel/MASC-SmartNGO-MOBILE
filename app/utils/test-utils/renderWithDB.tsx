jest.mock("storage/adapter.ts")
import DatabaseProvider from "@nozbe/watermelondb/DatabaseProvider"
import { render } from "@testing-library/react-native"
import { default as React } from "react"
import { database } from "storage/database"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProvidersWithDB({ children }: AllTheProvidersProps) {
    return (
        <DatabaseProvider database={database}>
            <ThemeProvider>{children}</ThemeProvider>
        </DatabaseProvider>
    )
}

export function renderWithDB<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
    return render(ui, { wrapper: AllTheProvidersWithDB, ...options })
}


export class DiagnosticError extends Error {
    name: string = "Diagnostic error"
  }
  