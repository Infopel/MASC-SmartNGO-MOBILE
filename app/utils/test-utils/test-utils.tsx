import { render } from "@testing-library/react-native"
import React from "react"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProviders({ children }: AllTheProvidersProps) {
  return (
    <ThemeProvider>{children}</ThemeProvider>
  )
}

function customRender<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
  return render(ui, { wrapper: AllTheProviders, ...options })
}

// re-export everything
export * from "@testing-library/react-native"
// override render method
export { customRender as render }

