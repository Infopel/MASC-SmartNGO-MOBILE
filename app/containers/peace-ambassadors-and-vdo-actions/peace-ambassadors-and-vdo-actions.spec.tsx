import { fireEvent, render } from "utils/test-utils/test-utils"
import * as React from "react"
import { PeaceAmbassadorsAndVdoActionsContainer } from "./peace-ambassadors-and-vdo-actions"

describe("<PeaceAmbassadorsAndVdoActionsContainer>",()=>{
 it("render", async function () {
    const component = <PeaceAmbassadorsAndVdoActionsContainer></PeaceAmbassadorsAndVdoActionsContainer>
    
    expect(render(component).toJSON()).toMatchSnapshot()
  })
})
  