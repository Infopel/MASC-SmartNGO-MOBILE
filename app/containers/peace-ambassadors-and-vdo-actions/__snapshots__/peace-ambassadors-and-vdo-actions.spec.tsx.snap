// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PeaceAmbassadorsAndVdoActionsContainer> render 1`] = `
<View
  dataSet={Object {}}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    contentContainerStyle={
      Array [
        Object {},
        Object {
          "dataSet": Object {},
        },
      ]
    }
    dataSet={Object {}}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={Object {}}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View />
  </RCTScrollView>
  <View
    dataSet={Object {}}
    style={Object {}}
  />
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    colorScheme="primary"
    dataSet={Object {}}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onHoverIn={[Function]}
    onHoverOut={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#d37b16",
        "borderRadius": 9999,
        "bottom": 16,
        "elevation": 8,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
        "position": "absolute",
        "right": 16,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 4.65,
        "zIndex": 20,
      }
    }
  >
    <View
      dataSet={Object {}}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
      test={true}
    >
      <View
        dataSet={Object {}}
        style={Object {}}
      >
        <Text
          dataSet={Object {}}
          style={
            Object {
              "backgroundColor": undefined,
              "color": "#fafafa",
              "fontFamily": undefined,
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 21,
              "textDecorationLine": undefined,
            }
          }
        >
          Add action
        </Text>
      </View>
    </View>
  </View>
</View>
`;
